### PR TITLE
Add configuration parameters for e2e multi-vc test

### LINKED ISSUE #412: K8s e2e tests configuration for multi-vc
Currently, we accept environment variables for single VC, we need to change to multi-vc.

### PR DESCRIPTION
**Pull request by abrarshivani**
Currently, to test e2e VCP we accept environment variables for single VC, we need to change to multi-vc. (Issue #412). This PR does the following things:
1. Fixes #412 by accepting the 'vsphere.conf'
2. Moves all env variable used for some tests to a single file. 

Found new issue while working on this, filed here: (https://github.com/vmware/kubernetes/issues/419)

Note:
1. New env variable added for accepting file location of the conf file:  "VSPHERE_CONF_FILE"
2. You need enter some random vm-uuid in the configuration file. (Issue: https://github.com/vmware/kubernetes/issues/419)

### CODE REVIEW

File: test/e2e/storage/vsphere_volume_perf.go [hunk 1]
```diff
@@ -69,25 +66,20 @@ var _ = SIGDescribe("vcp-performance [Feature:vsphere]", func() {
...
-		iterations, err = strconv.Atoi(iterationsStr)
+		Expect(VCPPerfIterations).NotTo(BeEmpty(), "ENV VCP_PERF_ITERATIONS is not set")
+		iterations, err = strconv.Atoi(VCPPerfIterations)
 		Expect(err).NotTo(HaveOccurred(), "Error Parsing VCP_PERF_ITERATIONS")
 
-		policyName = os.Getenv("VSPHERE_SPBM_GOLD_POLICY")
-		datastoreName = os.Getenv("VSPHERE_DATASTORE")
-		Expect(policyName).NotTo(BeEmpty(), "ENV VSPHERE_SPBM_GOLD_POLICY is not set")
-		Expect(datastoreName).NotTo(BeEmpty(), "ENV VSPHERE_DATASTORE is not set")
+		Expect(SPBMGoldPolicyName).NotTo(BeEmpty(), "ENV VSPHERE_SPBM_GOLD_POLICY is not set")
```
Comment (divyenpatel) [inline test/e2e/storage/vsphere_volume_perf.go]: Lets change SPBMGoldPolicyName -> SPBMPolicyName

File: test/e2e/storage/vsphere_volume_vsan_policy.go [hunk 2]
```diff
@@ -207,10 +203,9 @@ var _ = SIGDescribe("Storage Policy Based Volume Provisioning [Feature:vsphere]"
 	})
 
 	It("verify an existing and compatible SPBM policy is honored for dynamically provisioned pvc using storageclass", func() {
-		By(fmt.Sprintf("Invoking Test for SPBM policy: %s", os.Getenv("VSPHERE_SPBM_GOLD_POLICY")))
-		goldPolicy := os.Getenv("VSPHERE_SPBM_GOLD_POLICY")
-		Expect(goldPolicy).NotTo(BeEmpty())
-		scParameters[SpbmStoragePolicy] = goldPolicy
+		By(fmt.Sprintf("Invoking Test for SPBM policy: %s",))
```
Comment (divyenpatel) [inline test/e2e/storage/vsphere_volume_vsan_policy.go]: By(fmt.Sprintf("Invoking Test for SPBM policy: %s",)) => Need to pass policy name here for console printing.

File: test/e2e/storage/vsphere_scale.go [hunk 3]
```diff
@@ -77,27 +74,22 @@ var _ = SIGDescribe("vcp at scale [Feature:vsphere] ", func() {
 		nodeVolumeMapChan = make(chan map[string][]string)
 
 		// Read the environment variables
-		volumeCountStr := os.Getenv("VCP_SCALE_VOLUME_COUNT")
-		Expect(volumeCountStr).NotTo(BeEmpty(), "ENV VCP_SCALE_VOLUME_COUNT is not set")
-		volumeCount, err = strconv.Atoi(volumeCountStr)
+		Expect(VCPScaleVolumeCount).NotTo(BeEmpty(), "ENV VCP_SCALE_VOLUME_COUNT is not set")
```
Comment (divyenpatel) [inline test/e2e/storage/vsphere_scale.go]: I think VCPScaleVolumeCount, VCPScaleVolumesPerPod, VCPScaleInstances, SPBMPolicyName and StorageClassDatastoreName is not defined. 
Can you compile the code changes and make sure?

File: test/e2e/storage/vsphere_volume_cluster_ds.go [hunk 4]
```diff
@@ -75,7 +72,7 @@ var _ = SIGDescribe("Volume Provisioning On Clustered Datastore [Feature:vsphere
 		volumeOptions := new(vclib.VolumeOptions)
 		volumeOptions.CapacityKB = 2097152
 		volumeOptions.Name = "e2e-vmdk-" + namespace
-		volumeOptions.Datastore = clusterDatastore
+		volumeOptions.Datastore = VCPClusterDatastore
```
Comment (divyenpatel) [inline test/e2e/storage/vsphere_volume_cluster_ds.go]: VCPClusterDatastore is not defined.

File: test/e2e/storage/vsphere_volume_disksize.go [hunk 5]
```diff
@@ -57,7 +56,7 @@ var _ = SIGDescribe("Volume Disk Size [Feature:vsphere]", func() {
 		client = f.ClientSet
 		namespace = f.Namespace.Name
 		scParameters = make(map[string]string)
-		datastore = os.Getenv("VSPHERE_DATASTORE")
+		datastore = StorageClassDatastoreName
```
Comment (divyenpatel) [inline test/e2e/storage/vsphere_volume_disksize.go]: StorageClassDatastoreName not defined or not imported?

File: test/e2e/storage/vsphere_volume_perf.go [hunk 6]
```diff
@@ -69,25 +66,20 @@ var _ = SIGDescribe("vcp-performance [Feature:vsphere]", func() {
 		namespace = f.Namespace.Name
 
 		// Read the environment variables
-		volumeCountStr := os.Getenv("VCP_PERF_VOLUME_COUNT")
-		Expect(volumeCountStr).NotTo(BeEmpty(), "ENV VCP_PERF_VOLUME_COUNT is not set")
-		volumeCount, err = strconv.Atoi(volumeCountStr)
+		Expect(VCPPerfVolumeCount).NotTo(BeEmpty(), "ENV VCP_PERF_VOLUME_COUNT is not set")
```
Comment (divyenpatel) [inline test/e2e/storage/vsphere_volume_perf.go]: VCPPerfVolumeCount, VCPPerfVolumesPerPod, VCPPerfIterations, SPBMGoldPolicyName, StorageClassDatastoreName not defined.

File: test/e2e/storage/vsphere_volume_placement.go [hunk 7]
```diff
@@ -224,7 +223,7 @@ var _ = SIGDescribe("Volume Placement", func() {
 		volumeOptions = new(vclib.VolumeOptions)
 		volumeOptions.CapacityKB = 2097152
 		volumeOptions.Name = "e2e-vmdk-" + strconv.FormatInt(time.Now().UnixNano(), 10)
-		volumeOptions.Datastore = os.Getenv("VSPHERE_SECOND_SHARED_DATASTORE")
+		volumeOptions.Datastore = SecondSharedDatastore
```
Comment (divyenpatel) [inline test/e2e/storage/vsphere_volume_placement.go]: SecondSharedDatastore is not imported or declared.

**Comment by abrarshivani** [comment]
@divyenpatel Good Catch! I added those variables in vsphere_common.go file which is not added to git. Will add that file to git and push the changes

File: test/e2e/storage/vsphere_volume_perf.go (hunk 1 above)
Comment (abrarshivani) [reply test/e2e/storage/vsphere_volume_perf.go]: We have already have variable SPBMPolicyName

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go [hunk 8]
```diff
@@ -50,59 +52,31 @@ const (
...
-		cfg.Global.InsecureFlag = true
-	}
-	cfg.Workspace.VCenterIP = cfg.Global.VCenterIP
-	cfg.Workspace.Datacenter = cfg.Global.Datacenters
-	cfg.Workspace.DefaultDatastore = cfg.Global.DefaultDatastore
-	cfg.Workspace.Folder = cfg.Global.WorkingDir
-	return &cfg
+func getVSphereConfig() (*VSphereConfig, error) {
+	if ConfFileLocation == "" {
+		return nil, fmt.Errorf("Env variable 'VSPHERE_CONF_FILE' is empty.")
```
Comment (shaominchen) [inline pkg/cloudprovider/providers/vsphere/vsphere_util.go]: "is empty" => "is not set"

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go [hunk 9]
```diff
@@ -526,3 +491,30 @@ func (vs *VSphere) checkDiskAttached(ctx context.Context, nodes []k8stypes.NodeN
 	}
 	return nodesToRetry, nil
 }
+
+func (vs *VSphere) IsDummyVMPresent(vmName string) (bool, error) {
```
Comment (shaominchen) [inline pkg/cloudprovider/providers/vsphere/vsphere_util.go]: Let's add an empty line in between separate semantic blocks, e.g. line 499/500, 503/504, 507/508, etc. to improve code readability a little bit.

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go (hunk 8 above)
Comment (abrarshivani) [reply pkg/cloudprovider/providers/vsphere/vsphere_util.go]: Done

File: pkg/cloudprovider/providers/vsphere/vsphere_util.go (hunk 9 above)
Comment (abrarshivani) [reply pkg/cloudprovider/providers/vsphere/vsphere_util.go]: Done

File: test/e2e/storage/vsphere_volume_perf.go (hunk 1 above)
Comment (abrarshivani) [reply test/e2e/storage/vsphere_volume_perf.go]: Done removed this variable

**Comment by prashima** [comment]
Already merged.

**Pull request closed without merging**